gm4_storage_drawers/data/storage_drawers/functions/all/upgrade_diamond.mcfunction

```diff
@@ -0,0 +1,6 @@
+#run from detection_master_main
+#@s = as and at drawer with acceptable upgrade
+
+scoreboard players set @s gm4_sd_max 6144
+kill @e[type=item, limit=1, sort=nearest, distance=..1, nbt={Item: {tag: {CustomModelData:4911315144, display: {Name: '{"text":"Storage Upgrade: Diamond"}'}}}}]
```

Comment:
I'm pretty sure this CustomModelData exceeds the max int number making this invalid? I haven't properly checked the other data values you've used but I think this is going to be an issue with some of the others at a glance. The int limit is 2147483647 but I've heard there's a bug in CustomModelData that stops larger numbers from working so it might be worth using lower numbers as your ids.